### PR TITLE
Support asymmetric encryption keys

### PR DESCRIPTION
**Pull request by harti2006**
Fixes https://github.com/zalando/spring-cloud-config-aws-kms/issues/82
Adds support for asymmetric KMS keys.

Unfortunately, the encryption algorithm feature is quite new and not available in the AWS Java SDK version that Spring Cloud dependency management ships with (current version `Hoxton.RELEASE`). Users who want to use the new feature, need to upgrade their aws java sdk manually.

- [x] enable encryption and decryption with asymmetric keys, given global encryption algorithm and kmsKeyId
- [x] enable decryption with asymmetric keys, given local kmsKeyId and encryptionAlgorithm
    `{cipher}[keyId=1234567890,algorithm=RSAES_OAEP_SHA_256]CiA47hYvQq==` 
- [x] make it backward compatible (for users that don't want to deal with AWS Java SDK upgrade)
- [x] create test
- [x] create documentation

### CODE REVIEW

**Comment by balazsmaria** [comment]
any timeline when this can be merged?

**Comment by harti2006** [comment]
@balazsmaria Sorry, I'm quite busy right now. I hope to find some spare time in April to get this done.

**Comment by balazsmaria** [comment]
what's missing? can I help you?

**Comment by harti2006** [comment]
There are not enough tests to cover all aspects of the new feature. I even refactored the project into multi-modules to be able to test the code with different set of dependencies, i.e. one test with recent version of AWS SDK that supports asymmetric keys, another test with old (Spring Cloud default) version of AWS SDK, etc.
Also docs are missing.
It would already help a lot, if you could check out the branch, build it and try the snapshot version locally in your project to confirm that the API makes sense and everything works as expected

File: spring-cloud-config-aws-kms/src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsTextEncryptor.java [hunk 1]
```diff
@@ -61,23 +91,47 @@ public String decrypt(final String encryptedText) {
...
+                return new String(bytes);
             }
         } else {
             return EMPTY_STRING;
         }
     }
+
+    private void checkAlgorithm(String algorithm) {
+        if (isAsymmetricEncryption(algorithm) && !IS_ALGORITHM_AVAILABLE) {
+            log.warn("Non-symmetric encryption '{}' has been configured," +
```
Comment (mrandi) [inline spring-cloud-config-aws-kms/src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsTextEncryptor.java:128]: Asymmetric maybe?

File: spring-cloud-config-aws-kms/src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsTextEncryptorOptions.java [hunk 2]
```diff
@@ -8,24 +8,40 @@
 
     private final OutputMode outputMode;
 
-    KmsTextEncryptorOptions(OutputMode outputMode) {
+    private final String keyId;
+
+    private final String encryptionAlgorithm;
+
+    KmsTextEncryptorOptions(OutputMode outputMode, String keyId, String encryptionAlgorithm) {
```
Comment (mrandi) [inline spring-cloud-config-aws-kms/src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsTextEncryptorOptions.java:15]: what about using a builder?

**Comment by mrandi** [comment]
👍

**Comment by harti2006** [comment]
:+1:

**Pull request merged**